code/lib/create-storybook/src/bin/index.ts

```diff
@@ -1,28 +1,60 @@
-import { versions } from 'storybook/internal/common';
+/* eslint-disable no-underscore-dangle */
 import { addToGlobalContext } from 'storybook/internal/telemetry';
 
 import { program } from 'commander';
-import { findPackageSync } from 'fd-package-json';
-import invariant from 'tiny-invariant';
 
+import { version } from '../../package.json';
 import type { CommandOptions } from '../generators/types';
 import { initiate } from '../initiate';
+import { modernInputs } from './modernInputs';
 
-addToGlobalContext('cliVersion', versions.storybook);
+const IS_NON_CI = process.env.CI !== 'true';
+const IS_NON_STORYBOOK_SANDBOX = process.env.IN_STORYBOOK_SANDBOX !== 'true';
 
-const pkg = findPackageSync(__dirname);
-invariant(pkg, 'Failed to find the closest package.json file.');
+addToGlobalContext('cliVersion', version);
 
-program
-  .name('Initialize Storybook into your project.')
+const createStorybookProgram = program.name('Initialize Storybook into your project.');
+
+const modernProgram = Object.entries(modernInputs.shape).reduce((acc, [key, schema]) => {
```

Comment:
It's not immediately obvious what this reduce is for. Could you add a comment?

Reply:
This converts the zod runtime interface to a `program.options()`-chain for `commander`